test/TesterInternal/TimeoutTests.cs

```diff
@@ -84,5 +85,37 @@ public void Timeout_LongMethod()
             Assert.True(stopwatch.Elapsed <= timeout.Multiply(0.1), "Waited longer than " + timeout.Multiply(0.1) + ". Waited " + stopwatch.Elapsed);
             Assert.True(promise.Status == TaskStatus.Faulted);
         }
+
+
+        [Fact, TestCategory("SlowBVT")]
+        public async Task CallThatShouldHaveBeenDroppedNotExecutedTest()
+        {
+            var responseTimeout = TimeSpan.FromSeconds(2);
+            this.runtimeClient.SetResponseTimeout(responseTimeout);
+
+            var target = Client.GetGrain<ILongRunningTaskGrain<int>>(Guid.NewGuid());
+
+            // First call should be successful, but client will not receive the response
+            var firstCall = target.LongRunningTask(1, responseTimeout + TimeSpan.FromSeconds(5));
+            await Task.Delay(TimeSpan.FromMilliseconds(100));
+            // Second call should be dropped by the silo
+            var secondCall = target.LongRunningTask(2, TimeSpan.Zero);
+
+            try
+            {
+                await Assert.ThrowsAsync<TimeoutException>(() => firstCall);
+                await Assert.ThrowsAsync<TimeoutException>(() => secondCall);
+            }
+            catch
+            {
+                output.WriteLine(firstCall.IsFaulted ? $"firstCall: faulted" : $"firstCall: {firstCall.Result}");
+                output.WriteLine(secondCall.IsFaulted ? $"secondCall: faulted" : $"secondCall: {secondCall.Result}");
+                throw;
+            }
+
+            await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
nit: I'd make this `TimeSpan.FromSeconds(5)` a variable and use it here and in firstCall, otherwise it's not really clear why you are waiting.